mono/metadata/sgen-tarjan-bridge.c

```diff
@@ -115,7 +121,7 @@ typedef struct _ScanData {
 	// Tarjan algorithm index (order visited)
 	int index;
 	// Tarjan index of lowest-index object known reachable from here
-	int low_index : 27;
+	signed low_index : 27;
```

Comment:
I looked into it. `signed` is the default for GCC, Clang, MSVC, and ICC, so it looks like we’re safe. We could add `-fsigned-bitfields` for GCC and Clang, but it’s the default; Clang doesn’t even support `-funsigned-bitfields`, actually.

I would be more concerned about using `gboolean flag : 1;` and having a true value get silently truncated to false.